custom_components/localtuya/__init__.py

```diff
@@ -112,6 +112,19 @@ def __init__(self, device, config_entry, dps_id, **kwargs):
         self._dps_id = dps_id
         self._status = None
 
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                ("LocalTuya", f"local_{self._device.unique_id}")
+            },
+            "name": self._device._friendly_name,
```

Comment:
Friendly name is in config entry:

```
self._config_entry.data[CONF_FRIENDLY_NAME]
```
And revert change in the cache until we remove it.